crates/rune/src/compile/context.rs

```diff
@@ -881,12 +889,16 @@ impl Context {
         &mut self,
         hash: Hash,
         handler: &Arc<FunctionHandler>,
+        deprecation: &Option<Box<str>>,
```

Comment:
Nit: Should probably prefer using `Option<&str>`. There's no benefit for `try_clone` to know it's a `Box<str>`. And we have [`TryToOwned`](https://docs.rs/rune-alloc/latest/rune_alloc/borrow/trait.TryToOwned.html#impl-TryToOwned-for-str).